### PR TITLE
Introduce phpstan extension with UseStrictTypesForNewClassesRule

### PR DESCRIPTION
**Pull request by matks**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Introduce [phpstan extension](https://github.com/matks/phpstan-prestashop/) with UseStrictTypesForNewClassesRule that will check whether there are new files that do not use strict types.
| Type?             | new feature
| Category?         | TE
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | Fixes 1 item of https://github.com/PrestaShop/PrestaShop/issues/22728
| How to test?      | PhpStan CI should pass
| Possible impacts? | 

**If this PR is approved, what I can do is transfer [phpstan-prestashop/](https://github.com/matks/phpstan-prestashop/) repository to the opensource organization and update Packagist and the PR accordingly.**

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/22721)

### CODE REVIEW

File: src/Adapter/Product/SpecificPrice/CommandHandler/AddProductSpecificPriceHandler.php [hunk 1]
```diff
@@ -24,6 +24,8 @@
  * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
  */
 
+declare(strict_types=1);
```
Comment (matks) [inline src/Adapter/Product/SpecificPrice/CommandHandler/AddProductSpecificPriceHandler.php:27]: Found thanks to the rule 🎉

File: src/Core/Domain/Product/Exception/ProductCustomizationNotFoundException.php [hunk 2]
```diff
@@ -24,6 +24,8 @@
  * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
  */
 
+declare(strict_types=1);
```
Comment (matks) [inline src/Core/Domain/Product/Exception/ProductCustomizationNotFoundException.php:27]: Found thanks to the rule 🎉

**Comment by matks** [comment]
WOOOPS I pushed on upstream instead of my fork 😱

**Comment by kpodemski** [comment]
Nice 👍 Good idea to have it in PrestaShop org, I'll give my accept after moving it :D

**Comment by PierreRambaud** [comment]
I approve the PR but we will have to close it. Please use your own fork and move the PHPStan extension into PrestaShop org :)

**Comment by matks** [comment]
New PR https://github.com/PrestaShop/PrestaShop/pull/22764/

**Pull request closed without merging**